linkerd/addr/src/lib.rs

```diff
@@ -16,7 +17,7 @@ pub enum Addr {
 
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
 pub struct NameAddr {
-    name: Name,
+    name: Arc<Name>,
```

Comment:
nit, take it or leave it: this change could probably land on `main` in a totally separate branch?